Lib/traceback.py

```diff
@@ -904,7 +913,17 @@ def format_exception_only(self, *, show_group=False, _depth=0):
             stype = smod + '.' + stype
 
         if not issubclass(self.exc_type, SyntaxError):
-            yield indent + _format_final_exc_line(stype, self._str)
+            if _depth > 0:
+                # Nested exceptions needs correct handling of multiline messages.
+                formatted = _format_final_exc_line(
+                    stype, self._str, insert_final_newline=False,
+                ).split('\n')
+                yield from [
+                    indent + l + '\n'
+                    for l in formatted
+                ]
+            else:
+                yield _format_final_exc_line(stype, self._str)
```

Comment:
No, because 3.12 does not have `show_group` / `_depth` at all: https://github.com/python/cpython/blob/307ca78d7f97e6942c64383464a7de073d8b6e42/Lib/traceback.py#L865-L892

And without it, the result is identical with and without this fix.